scripts/compile-test/Archlinux

```diff
@@ -20,4 +20,4 @@ RUN pacman -Sy --noconfirm gcc extra-cmake-modules threadweaver kconfigwidgets k
 
 ADD . /opt/hotspot/
 
-RUN cd hotspot && mkdir build && cd build && cmake .. && make
+RUN cd hotspot && mkdir build && cd build && cmake -DBUILD_WITH_ASAN=ON .. && make
```

Comment:
let's introduce `CMakePresets.json` and leverage it here with `cmake --build --preset dev-asan`

```
{
  "version": 2,
  "configurePresets": [
    {
      "name": "dev",
      "displayName": "dev",
      "description": "A basic preset for developers",
      "generator": "Ninja",
      "binaryDir": "${sourceDir}/build-${presetName}",
      "cacheVariables": {
        "CMAKE_BUILD_TYPE": "Debug",
        "CMAKE_EXPORT_COMPILE_COMMANDS": "ON",
      }
    },
    {
      "name": "dev-asan",
      "displayName": "dev with sanitizers",
      "description": "A basic preset for developers with asan and ubsan enabled",
      "cacheVariables": {
        "CMAKE_CXX_FLAGS_DEBUG": "-g -fsanitize=address,undefined",
        "CMAKE_C_FLAGS_DEBUG": "-g -fsanitize=address,undefined"
      },
      "inherits": [
        "dev"
      ]
    }
  ],
  "buildPresets": [
    {
      "name": "dev",
      "configurePreset": "dev"
    },
    {
      "name": "dev-asan",
      "configurePreset": "dev-asan"
    }
  ]
}
```